src/Polly.Shared/NoOp/NoOpEngineAsync.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Polly.NoOp
+{
+    internal static partial class NoOpEngine
+    {
+        internal static async Task<TResult> ImplementationAsync<TResult>(Func<CancellationToken, Task<TResult>> action,
+            CancellationToken cancellationToken)
+        {
+            return await action(cancellationToken);
```

Comment:
@lakario  Additional tiny spot: we want to add a `.ConfigureAwait(continueOnCapturedContext)` here.  We place these on every `await`, to guarantee we honour the user's choice of `continueOnCapturedContext` all through an async call.

(I've been using [Resharper plugin](https://resharper-plugins.jetbrains.com/packages/ConfigureAwaitChecker.v9) to monitor these ; we are also considering adding the [relevant Roslyn analyzer](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Microsoft.ApiDesignGuidelines.Analyzers.md#ca2007-do-not-directly-await-a-task) in the CI build)